tools/src/main/java/org/apache/kafka/tools/StateChangeLogMerger.java

```diff
@@ -0,0 +1,329 @@
+import org.apache.kafka.server.util.ToolsUtils;
+
+import java.io.File;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.io.Serializable;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Date;
+import java.util.Arrays;
+import java.util.List;
+import java.util.PriorityQueue;
+import java.util.Scanner;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * A utility that merges the state change logs (possibly obtained from different brokers and over multiple days).
+ *
+ * This utility expects at least one of the following two arguments -
+ * 1. A list of state change log files
+ * 2. A regex to specify state change log file names.
+ *
+ * This utility optionally also accepts the following arguments -
+ * 1. The topic whose state change logs should be merged
+ * 2. A list of partitions whose state change logs should be merged (can be specified only when the topic argument
+ * is explicitly specified)
+ * 3. Start time from when the logs should be merged
+ * 4. End time until when the logs should be merged
+ */
+public class StateChangeLogMerger {
+    private static final String DATE_FORMAT = "yyyy-MM-dd HH:mm:ss,SSS";
+    private static final Pattern DATE_PATTERN = Pattern.compile("([0-9]{4}-[0-9]{2}-[0-9]{2} [0-9]{2}:[0-9]{2}:[0-9]{2},[0-9]{3}){1}");
+    private static final Pattern TOPIC_PART_PATTERN = Pattern.compile("\\[(" + Topic.LEGAL_CHARS + "+),([0-9]+)\\]");
+
+    private static List<String> files;
+    private static String topic;
+    private static List<Integer> partitions;
+    private static Date startDate;
+    private static Date endDate;
+
+    public static void main(String[] args) {
+        try {
+            StateChangeLogMergerOptions options = new StateChangeLogMergerOptions(args);
+            if (CommandLineUtils.isPrintHelpNeeded(options)) {
+                CommandLineUtils.printUsageAndExit(options.parser,
+                    "A tool for merging the log files from several brokers to reconstruct a unified history of what happened.");
+                return;
+            }
+            if (CommandLineUtils.isPrintVersionNeeded(options)) {
+                CommandLineUtils.printVersionAndExit();
+                return;
+            }
+
+            if ((!options.hasFiles() && !options.hasRegex()) || (options.hasFiles() && options.hasRegex())) {
+                CommandLineUtils.printUsageAndExit(options.parser,
+                    String.format("Provide arguments to exactly one of the two options \"%s\" or \"%s\"", options.filesOpt, options.regexOpt));
+                return;
+            }
+            if (options.hasPartitions() && !options.hasTopic()) {
```

Comment:
Similiar to the previous comment can't we use `CommandLineUtils.checkRequiredArgs`.